src/general/system.jl

```diff
@@ -151,3 +151,17 @@ end
 
 # Only for systems requiring a mandatory callback
 reset_callback_flag!(system) = system
+
+# Assuming a constant particle spacing one can calculate the number of neighbors within the
+# compact support for an undisturbed particle distribution.
+function number_density(::Val{D}, particle_spacing, compact_support) where {D}
+    throw(ArgumentError("Unsupported dimension: $D"))
+end
+
+@inline function number_density(::Val{2}, particle_spacing, compact_support)
+    return floor(Int64, pi * compact_support^2 / particle_spacing^2)
```

Comment:
```suggestion
    return floor(Int, pi * compact_support^2 / particle_spacing^2)
```
Please use `Int` when `Int64` is not explicitly necessary. Also everywhere else in this PR.